db/src/db.rs

```diff
@@ -457,17 +462,149 @@ pub enum SearchType {
     Inexact,
 }
 
-impl DB {
-
+/// `MentatStoring` will be the trait that encapsulates the storage layer.  It is consumed by the
+/// transaction processing layer.
+///
+/// Right now, the only implementation of `MentatStoring` is the SQLite-specific SQL schema.  In the
+/// future, we might consider other SQL engines (perhaps with different fulltext indexing), or
+/// entirely different data stores, say ones shaped like key-value stores.
+pub trait MentatStoring {
     /// Given a slice of [a v] lookup-refs, look up the corresponding [e a v] triples.
     ///
     /// It is assumed that the attribute `a` in each lookup-ref is `:db/unique`, so that at most one
     /// matching [e a v] triple exists.  (If this is not true, some matching entid `e` will be
     /// chosen non-deterministically, if one exists.)
     ///
     /// Returns a map &(a, v) -> e, to avoid cloning potentially large values.  The keys of the map
-    /// are exactly those (a, v) pairs that have an assertion [e a v] in the datom store.
-    pub fn resolve_avs<'a>(&self, conn: &rusqlite::Connection, avs: &'a [&'a AVPair]) -> Result<AVMap<'a>> {
+    /// are exactly those (a, v) pairs that have an assertion [e a v] in the store.
+    fn resolve_avs<'a>(&self, avs: &'a [&'a AVPair]) -> Result<AVMap<'a>>;
+
+    /// Begin (or prepare) the underlying storage layer for a new Mentat transaction.
+    ///
+    /// Use this to create temporary tables, prepare indices, set pragmas, etc, before the initial
+    /// `insert_non_fts_searches` invocation.
+    fn begin_transaction(&self) -> Result<()>;
+
+    // TODO: this is not a reasonable abstraction, but I don't want to really consider non-SQL storage just yet.
+    fn insert_non_fts_searches<'a>(&self, entities: &'a [ReducedEntity], search_type: SearchType) -> Result<()>;
```

Comment:
This function name bothers me in the same way that it bothers me when people talk about PIN numbers. Which is to say it's not actually annoying, but is distracting, because it's not what I want to be thinking about.

You altered a function with a similar name earlier, but then you weren't adding the function so I let you off ;-)